src/PolyType.Roslyn/SourceWriter.cs

```diff
@@ -10,10 +12,13 @@ namespace PolyType.Roslyn;
 /// <summary>
 /// A utility class for generating indented source code.
 /// </summary>
-public sealed class SourceWriter
+public sealed class SourceWriter : IDisposable
 {
-    private readonly StringBuilder _sb = new();
+    private static readonly string s_newLine = "\r\n";
```

Comment:
This should probably defer to Environment.NewLine. Do whatever StringBuilder is doing already for newline handling.